### ISSUE TITLE
Ghost test for the recursor is broken in master

### ISSUE THREAD

**Issue by pieterlexis** [PowerDNS/pdns#2741]
Also, `regression-tests.recursor/runtests` doesn't exit with a non-zero exit code on failure.

```
13 out of 15 (86.66%) tests passed, 0 were skipped
+context=recursor ../regression-tests/toxml
+grep . answer-in-local-auth/diff auth-zones/diff cname-to-a-nxdomain/diff cross-zone-cname-bogus-nxdomain/diff direct-cname-to-nxdomain/diff direct-cname/diff ghost-1/diff ghost-2/diff hijack-1/diff in-zone-cname-bogus-nxdomain/diff lame-noerror/diff simple-a/diff simple-cname-to-a/diff truncate-empty/diff txt-escaping/diff
ghost-1/diff:--- ./ghost-1/expected_result  2015-09-10 11:26:37.800505049 +0000
ghost-1/diff:+++ ./ghost-1/real_result  2015-09-10 11:47:10.537085042 +0000
ghost-1/diff:@@ -7,9 +7,9 @@
ghost-1/diff: 0 c.www.1.ghost.example.net.  IN  A   3600    192.0.2.7
ghost-1/diff: Rcode: 0, RD: 1, QR: 1, TC: 0, AA: 0, opcode: 0
ghost-1/diff: Reply to question for qname='c.www.1.ghost.example.net.', qtype=A
ghost-1/diff:-1 ghost.example.net.  IN  SOA 3600    ns.example.net. hostmaster.example.net. 1 3600 1800 1209600 300
ghost-1/diff:-Rcode: 3, RD: 1, QR: 1, TC: 0, AA: 0, opcode: 0
ghost-1/diff:+0 d.www.1.ghost.example.net.  IN  A   3600    192.0.2.7
ghost-1/diff:+Rcode: 0, RD: 1, QR: 1, TC: 0, AA: 0, opcode: 0
ghost-1/diff: Reply to question for qname='d.www.1.ghost.example.net.', qtype=A
ghost-1/diff:-1 ghost.example.net.  IN  SOA 3600    ns.example.net. hostmaster.example.net. 1 3600 1800 1209600 300
ghost-1/diff:-Rcode: 3, RD: 1, QR: 1, TC: 0, AA: 0, opcode: 0
ghost-1/diff:+0 e.www.1.ghost.example.net.  IN  A   3600    192.0.2.7
ghost-1/diff:+Rcode: 0, RD: 1, QR: 1, TC: 0, AA: 0, opcode: 0
ghost-1/diff: Reply to question for qname='e.www.1.ghost.example.net.', qtype=A
ghost-2/diff:--- ./ghost-2/expected_result  2015-09-10 11:26:37.800505049 +0000
ghost-2/diff:+++ ./ghost-2/real_result  2015-09-10 11:47:33.617766695 +0000
ghost-2/diff:@@ -7,9 +7,9 @@
ghost-2/diff: 0 c.www.2.ghost.example.net.  IN  A   3600    192.0.2.8
ghost-2/diff: Rcode: 0, RD: 1, QR: 1, TC: 0, AA: 0, opcode: 0
ghost-2/diff: Reply to question for qname='c.www.2.ghost.example.net.', qtype=A
ghost-2/diff:-1 ghost.example.net.  IN  SOA 3600    ns.example.net. hostmaster.example.net. 1 3600 1800 1209600 300
ghost-2/diff:-Rcode: 3, RD: 1, QR: 1, TC: 0, AA: 0, opcode: 0
ghost-2/diff:+0 d.www.2.ghost.example.net.  IN  A   3600    192.0.2.8
ghost-2/diff:+Rcode: 0, RD: 1, QR: 1, TC: 0, AA: 0, opcode: 0
ghost-2/diff: Reply to question for qname='d.www.2.ghost.example.net.', qtype=A
ghost-2/diff:-1 ghost.example.net.  IN  SOA 3600    ns.example.net. hostmaster.example.net. 1 3600 1800 1209600 300
ghost-2/diff:-Rcode: 3, RD: 1, QR: 1, TC: 0, AA: 0, opcode: 0
ghost-2/diff:+0 e.www.2.ghost.example.net.  IN  A   3600    192.0.2.8
ghost-2/diff:+Rcode: 0, RD: 1, QR: 1, TC: 0, AA: 0, opcode: 0
ghost-2/diff: Reply to question for qname='e.www.2.ghost.example.net.', qtype=A
travis_time:end:0ff5f28c:start=1441885601456719240,finish=1441885658834316635,duration=57377597395
[0K
[32;1mThe command "./runtests" exited with 0.[0m
```

**Comment by rgacogne** [comment]
Fixed on master in #6458.

**Comment by rgacogne** [comment]
Fixed on rel/rec-4.1.x by #7303.
